### ISSUE TITLE
Extending the app API for social media hooks and promotion

### ISSUE THREAD

**Issue by ghost** [hivewallet/discussions#7]
I still hold the view that 2014/2015 will see holding and using Bitcoin as the #1 tech fashion trend. Therefore, I would like to add social media integrations for things like this to encourage self-promotion of Hive:

PUBLIC post via Twitter:
"I just bought X with Bitcoin using @hivewallet"

PUBLIC post via Twitter:
"I just sent $X with Bitcoin using @hivewallet"

PRIVATE post via Facebook(?):
"I just sent you $X worth of Bitcoin using @hivewallet" + blockchain.info txid link ... "and saved $25 by doing so"

etc.

It seems to me that it makes the most sense to create very basic API functions to do this, but of course the reason I post here is that this is up for debate. Any thoughts?

**Comment by mackuba** [comment]
This sounds more like something that should be integrated in Hive itself (like that first version on the transactions list now), not in the apps?...

**Comment by weilu** [comment]
Personally, I'm never a big fan of the social media sharing feature -- It's always kind of rude and intrusive. I don't trust apps to post ANYTHING on my behave. Not a feature I'd ever use.

**Comment by javgh** [comment]
I personally can't really see Bitcoin becoming much of a fashion trend. I think it will rather go in the direction of things like XMPP and SIP, in that it mostly works in the background and people use Bitcoin-based services, without even being aware of it most of the time. Except perhaps, that hopefully the expression "I'd like to pay that with Bitcoin" enters the vocabulary.

I don't "share" much stuff, so I find it hard to judge, whether more share-happy people will use this functionality, but it always seems to me, that financial related things are in general a more private thing and not necessarily something people would want to share. So I would worry that these features clutter up the app and give of a ridiculous vibe of web-2.0-share-all-the-things.

That said, I think it would be super-useful, if we manage to integrate some kind of viral/social component, that helps Hive spread. I still think that "Hive connect" is the best candidate for that.

Scenario: Hive user A just bought their first bitcoins and has them appear in the app. Exciting stuff, which he wants to share with the world. A "I just bought my first bitcoins" tweet might make sense here (so only the first time, not on any subsequent transactions, to keep it low-key). But I'd rather keep the interface clean and utilitarian and the Hive connect invite flow has the advantage, that it's both a perfectly legitimate feature, as well as a viral/social channel. So the user would send an Hive invite to a friend, who humors him and downloads the app, accepts the invitation and they appear in each other's contact lists. A couple of taps later some bitcoins are on the way to the friend. The user is excited about their first Bitcoin transaction, Hive got an additional download and since they are now in each other's contact lists, Hive and Bitcoin has a chance, of actually becoming a regularly used tool to settle tabs between these friends, rather than just a technology novelty, that is tried once and then forgotten.

So in summary, I'm for social features where they fulfill and actual additional function, rather than just purely "I want to tell the world what I'm doing".
